### PR TITLE
Enhancement to the sync_status module

### PR DESCRIPTION
**Pull request by ckulal**
Enhancement to the sync_status module: Include logic to check sync status from remote site
Enhancement to the user_create module: Include check of sync status post creating user in case of multi-site setup
Enhancement to the bucket_create module: Include check of sync status post creating bucket in case of multi-site setup

Logs:
http://magna002.ceph.redhat.com/ceph-qe-logs/Chaithra/sync_status_enhance/

### CODE REVIEW

File: rgw/v2/tests/s3_swift/reusable.py [hunk 1]
```diff
@@ -63,6 +63,26 @@ def create_bucket(bucket_name, rgw, user_info, location=None):
             raise TestExecError("bucket creation failed")
     else:
         raise TestExecError("bucket creation failed")
+
+    is_multisite = utils.is_cluster_multisite()
```
Comment (mkasturi18) [inline rgw/v2/tests/s3_swift/reusable.py:67]: @ckulal , can we have this as a method say **get_remote_ip**  in reusable.py, that we call the method here and as well in the user_create.py file.

**Review by mkasturi18 — CHANGES_REQUESTED** [review]

File: rgw/v2/tests/s3_swift/reusable.py (hunk 1 above)
Comment (ckulal) [reply rgw/v2/tests/s3_swift/reusable.py:67]: Done @mkasturi18 
logs: http://magna002.ceph.redhat.com/ceph-qe-logs/Chaithra/sync_status_enhance/retry/

**Comment by ckulal** [comment]
> @ckulal , I would suggest to test with the existing tests where we are doing a remote site sync (bucket stats/object download) etc.

Hi @viduship , This PR contains inclusion of sync status check from remote site post creating user and buckets. 
currently we are checking sync status only from existing site, so as per this PR only addition is sync check from remote site only in case of user and bucket creation.

**Review by anrao19 — APPROVED** [review]
LGTM

**Review by hmaheswa — APPROVED** [review]

**Comment by ckulal** [comment]
Hi @viduship , With the enhancement provided in this PR.

I tried execution of config: test_Mbuckets_with_Nobjects_0_shards_sync_test_haproxy.yaml
by commenting "time.sleep(1200)", https://github.com/red-hat-storage/ceph-qe-scripts/blob/master/rgw/v2/tests/s3_swift/reusable.py#L2117

i.e without waiting extra 20 minutes for sync lease period.

Pass log: http://magna002.ceph.redhat.com/ceph-qe-logs/Chaithra/sync_status_enhance/retry/without-sync-lease-wait

Hence with the changes given in this PR, In future we can try to avoid blindly sleep in some scenarios.
that might require some testing effor. I will create backlog for the same

**Review by viduship — APPROVED** [review]
looks good

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**